### PR TITLE
libs/libshout: Update to 2.4.1

### PR DESCRIPTION
**Pull request by diizzyy**
Maintainer: @psycho-nico 
Compile tested: Kirkwood, iomega iConnect, LEDE trunk
Run tested: N/A

Description:
Update libshout to 2.4.1
Remove speex support
Add mirror

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>

### CODE REVIEW

**Comment by diizzyy** [comment]
Compile tested with ices (incl fix in #3764 )

**Comment by thess** [comment]
Please update Makefile copyright date to 2017

File: libs/libshout/Makefile [hunk 1]
```diff
@@ -30,19 +31,19 @@ define Package/libshout/default
   CATEGORY:=Libraries
   TITLE:=Shoutcast client library
   URL:=http://www.icecast.org/download.php
-  DEPENDS:= +libvorbis +libvorbisidec +libpthread
+  DEPENDS:= +libvorbis +libvorbisidec +libpthread +libopenssl
```
Comment (thess) [inline libs/libshout/Makefile]: I would prefer to see OpenSSL as an optional configuration/variant.

**Comment by thess** [comment]
Why did you remove the patches?
120 - build with tremor
140 - cleaner build without doc.

**Comment by diizzyy** [comment]
Fixed, 140 doesn't apply and I can't really tell a difference on my rather slow buildbox anyway. In general I think we should avoid doing "unnecessary" patching as it makes updating more tedious than what it needs to be.

**Comment by hnyman** [comment]
> > Why did you remove the patches?
> > 120 - build with tremor
> > 140 - cleaner build without doc.
>
> 140 doesn't apply 

I think that you should fix the patch. "doesn't apply" is not that great answer as the reason is just that the context around the patch has changed in the version bump.

> In general I think we should avoid doing "unnecessary" patching as it makes updating more tedious than what it needs to be.

On the other hand, it makes no sense for us to build docs, examples or win32 specific stuff that does not go to final .ipk packages (or worse, goes in but is unusable in Linux or grows the package size unnecessarily). Additionally the docs/examples/tests may cause trouble/breakage as sometimes they require additional build tools (like latex etc.) or cause circular dependency due to tools used in tests. So it makes sense to disable the clearly unnecessary parts of the build.

But you are right, it is about balance between package maintenance work and compilation performance & simplicity.

> I can't really tell a difference on my rather slow buildbox anyway

That observation is likely correct. I tested building libshout and looks like the old 140 patch patches the wrong file...  Makefile.in is re-generated from Makefile.am during the make process before the actual compilation, so the patch has no practical effect and the "disabled" bits were compiled in any case :-(

Makefile.am should be patched instead. And that is much simpler. Below is example that works and disables building examples, doc and win32. I compared compile.log files and it really left those bits uncompiled...

```
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,7 +3,7 @@
 AUTOMAKE_OPTIONS = 1.6 foreign
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = include src examples doc win32
+SUBDIRS = include src
 
 EXTRA_DIST = INSTALL m4/shout.m4 m4/acx_pthread.m4 \
 	m4/ogg.m4 m4/vorbis.m4 m4/xiph_compiler.m4 m4/xiph_net.m4 \
```

Ps. 110-vorbis-c.patch has some fuzz. You should do ` make package/libshout/{clean,refresh} V=s ` to fix the last line.

**Comment by diizzyy** [comment]
Updated, thanks for the hand holding @hnyman :-)

**Comment by hnyman** [comment]
Anybody able to test this?
Willing to take over maintainership if @psycho-nico is MIA.

**Comment by diizzyy** [comment]
I can give it a go possibly later this week, on the other hand I think we can safely kill this along with ices as no sane person in 2017 would use OGG/Vorbis for streaming.

**Comment by thess** [comment]
Don't know much about the sanity of folks using ogg, but I would like to take over ices, libshout and icecast. I am a user of libshout and icecast and can test them with the shoutcast version of pianod. It would also be good to have SSL variants of these as well as non-SSL versions.

**Comment by diizzyy** [comment]
What else uses libshout? My point is that legacy usage --> MP3 and "anything" else uses AAC. Sorry if I've missed something but I did have quick look of libshout users and could only find ices.

**Comment by thess** [comment]
After cleaning up libshout (I will remove ogg). I intend to make the shoutcast version of pianod available.

**Comment by diizzyy** [comment]
Ahh, I see....
At some later point we should probably replace icecast(2) with icecast2-kh

**Comment by thess** [comment]
Sounds good to me - I try to get to it this week after I finish up with shairport-sync mbed tls support.

**Comment by thess** [comment]
Closed via 1c2107e46268d26b9c8794345cd1b59cf4e602a5

**Pull request closed without merging**